src/librbd/image/CloneRequest.h

```diff
@@ -0,0 +1,152 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_IMAGE_CLONE_REQUEST_H
+#define CEPH_LIBRBD_IMAGE_CLONE_REQUEST_H
+
+#include "common/WorkQueue.h"
+#include "include/rados/librados.hpp"
+#include "include/rbd_types.h"
+#include "cls/rbd/cls_rbd_types.h"
+#include "include/rbd/librbd.hpp"
+#include "librbd/ImageCtx.h"
+
+class Context;
+
+using librados::IoCtx;
+
+namespace librbd {
+namespace image {
+
+template <typename ImageCtxT = ImageCtx>
+class CloneRequest {
+public:
+  static CloneRequest *create(ImageCtx *p_imctx, IoCtx &c_ioctx, const std::string &c_name,
+			      ImageOptions c_options,
```

Comment:
CloneRequest need to pass an options to CreateRequest. So, we should keep an m_image_opts in CloneRequest, Because rbd-mirror is passing a local variable of options to CloneRequest, if we pass it by reference to CloneRequest, we will get an seg fault when defer it.